### PR TITLE
AM-385 delete login attempts username changed

### PR DESCRIPTION
**Pull request by ashraf706**

### CODE REVIEW

**Review by aurelien-pacaud — APPROVED** [review]
Tests OK & LGTM

**Review by remisultan — CHANGES_REQUESTED** [review]
One small concern here

File: gravitee-am-management-api/gravitee-am-management-api-service/src/main/java/io/gravitee/am/management/service/impl/AbstractUserService.java [hunk 1]
```diff
@@ -175,18 +182,25 @@ public Single<User> updateUsername(ReferenceType referenceType, String reference
...
                                         })
                                 ))
-                        .doOnSuccess(user1 -> auditService.report(AuditBuilder.builder(UserAuditBuilder.class).principal(principal).type(EventType.USERNAME_UPDATED).user(user1)))
+                        .doOnSuccess(user1 -> {
+                            auditService.report(AuditBuilder.builder(UserAuditBuilder.class).principal(principal).type(EventType.USERNAME_UPDATED).user(user1));
+                            if (DOMAIN.equals(referenceType)) {
+                                loginAttemptService.reset(createLoginAttemptCriteria(referenceId, oldUsername.get()))
+                                        .doOnError(error -> logger.info("Could not delete login attempt {}", error.getMessage()))
+                                        .subscribe();
+                            }
```
[the remark this answers is not in the document]
Comment (ashraf706) [reply gravitee-am-management-api/gravitee-am-management-api-service/src/main/java/io/gravitee/am/management/service/impl/AbstractUserService.java]: updated as suggested.

**Review by remisultan — APPROVED** [review]
LGTM

**Pull request merged**